### DISCUSSION TITLE
Setting connection-level SQL variables

### DISCUSSION THREAD

**Discussion by ceremonious** [prisma/prisma#18626] [Q&A]
We are using Prisma alongside Planetscale and are running into the following issue:

```
code = ResourceExhausted desc = Out of sort memory, consider increasing server sort buffer size (errno 1038) (sqlstate HY001)
```

In [this issue](https://github.com/planetscale/beam/issues/58), the suggested fix is calling this when the Prisma client is initialized: `set sort_buffer_size=2097152`

However, when I call `await prisma.$executeRaw'set sort_buffer_size=2097152'` after initializing the Prisma client, it seems like the value only gets set for one of the connections. 

If I run something like this:
```js
prisma.$queryRaw`SELECT @@global.sort_buffer_size as sort_buffer_size;`.then(
    (x) => console.log(`Sort buffer size: ${x[0]?.sort_buffer_size}`),
  );
prisma.$queryRaw`SELECT @@global.sort_buffer_size as sort_buffer_size;`.then(
  (x) => console.log(`Sort buffer size: ${x[0]?.sort_buffer_size}`),
);
prisma.$queryRaw`SELECT @@global.sort_buffer_size as sort_buffer_size;`.then(
  (x) => console.log(`Sort buffer size: ${x[0]?.sort_buffer_size}`),
);
```

The first query will print the correct value (2097152), but the other queries will not. I'm not sure why that's happening, but is there a way to make sure that every Prisma connection runs with the updated sort_buffer_size?

**Comment by ludralph** [comment]
Hi @ceremonious 👋 

Can you set the `sort_buffer_size` globally by doing this 
```js
await prisma.$executeRaw('SET GLOBAL sort_buffer_size = 2097152');
```

**Reply by ceremonious** [reply]
Unfortunately Planetscale does not allow setting variables at the global level, so that query results in no change being applied.

**Reply by nurul3101** [reply]
One alternative I can think of is to set the `sort_buffer_size` variable in [middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) before every query runs, but it might not be the best possible solution.

**Reply by nurul3101** [reply]
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by jborichevskiy** [comment]
Having this issue as well, curious if anyone has found a workaround? 

> One alternative I can think of is to set the sort_buffer_size variable in [middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) before every query runs, but it might not be the best possible solution.

Technically this works but in production ends up slowing down all the requests that don't need the expanded sort buffer.

**Reply by focux** [reply]
Hey! Did you find a way to do it?
